packages/scrollable-image/src/components/with-waypoints.js

```diff
@@ -0,0 +1,119 @@
+import PropTypes from 'prop-types'
+import React from 'react'
+import { Waypoint } from 'react-waypoint'
+
+const withWaypoints = WrappedComponent => {
+  class WithWaypoints extends React.PureComponent {
+    static propTypes = {
+      debug: PropTypes.bool,
+    }
+
+    static defaultProps = {
+      debug: false,
+    }
+    constructor(props) {
+      super(props)
+      this.state = {
+        isActive: false,
+        verticalDirection: null,
```

Comment:
`verticalDirection` -> `scrollDirection` ?

Or even `isScrollingFromTopToBottom`?
If so, we can remove `up` and `down`.